ragna/core/_rag.py

```diff
@@ -80,6 +83,7 @@ def chat(
         documents: Iterable[Any],
         source_storage: Union[Type[SourceStorage], SourceStorage],
         assistant: Union[Type[Assistant], Assistant],
+        embedding_model: Union[Type[GenericEmbeddingModel], GenericEmbeddingModel],
```

Comment:
We cannot make this required. Depending on the source storage, the user might not need to supply an embedding model.

```suggestion
        embedding_model: Optional[Union[Type[GenericEmbeddingModel], GenericEmbeddingModel]] = None,
```

And later in the `__init__` we can do something like

```python
if embedding_model is None and issubclass(source_storage.__ragna_input_type__, Embedding):
    raise RagnaException
```